core/common/src/main/java/alluxio/grpc/GrpcExceptionUtils.java

```diff
@@ -27,156 +22,13 @@
-        break;
-      case OK:
-        alluxioStatus = alluxio.exception.status.Status.OK;
-        break;
-      case OUT_OF_RANGE:
-        alluxioStatus = alluxio.exception.status.Status.OUT_OF_RANGE;
-        break;
-      case PERMISSION_DENIED:
-        alluxioStatus = alluxio.exception.status.Status.PERMISSION_DENIED;
-        break;
-      case RESOURCE_EXHAUSTED:
-        alluxioStatus = alluxio.exception.status.Status.RESOURCE_EXHAUSTED;
-        break;
-      case UNAUTHENTICATED:
-        alluxioStatus = alluxio.exception.status.Status.UNAUTHENTICATED;
-        break;
-      case UNAVAILABLE:
-        alluxioStatus = alluxio.exception.status.Status.UNAVAILABLE;
-        break;
-      case UNIMPLEMENTED:
-        alluxioStatus = alluxio.exception.status.Status.UNIMPLEMENTED;
-        break;
-      case UNKNOWN:
-        alluxioStatus = alluxio.exception.status.Status.UNKNOWN;
-        break;
-      default:
-        alluxioStatus = alluxio.exception.status.Status.UNKNOWN;
-    }
-
-    return AlluxioStatusException.from(alluxioStatus, e.getStatus().getDescription());
-  }
-
-  /**
-   * Convert from alluxio status exception to grpc exception.
-   *
-   * @param e the alluxio status exception
-   * @return the grpc status exception
-   */
-  public static StatusException toGrpcStatusException(AlluxioStatusException e) {
-    Status.Code code;
-    switch (e.getStatus()) {
-      case ABORTED:
-        code = Status.Code.ABORTED;
-        break;
-      case ALREADY_EXISTS:
-        code = Status.Code.ALREADY_EXISTS;
-        break;
-      case CANCELED:
-        code = Status.Code.CANCELLED;
-        break;
-      case DATA_LOSS:
-        code = Status.Code.DATA_LOSS;
-        break;
-      case DEADLINE_EXCEEDED:
-        code = Status.Code.DEADLINE_EXCEEDED;
-        break;
-      case FAILED_PRECONDITION:
-        code = Status.Code.FAILED_PRECONDITION;
-        break;
-      case INTERNAL:
-        code = Status.Code.INTERNAL;
-        break;
-      case INVALID_ARGUMENT:
-        code = Status.Code.INVALID_ARGUMENT;
-        break;
-      case NOT_FOUND:
-        code = Status.Code.NOT_FOUND;
-        break;
-      case OK:
-        code = Status.Code.OK;
-        break;
-      case OUT_OF_RANGE:
-        code = Status.Code.OUT_OF_RANGE;
-        break;
-      case PERMISSION_DENIED:
-        code = Status.Code.PERMISSION_DENIED;
-        break;
-      case RESOURCE_EXHAUSTED:
-        code = Status.Code.RESOURCE_EXHAUSTED;
-        break;
-      case UNAUTHENTICATED:
-        code = Status.Code.UNAUTHENTICATED;
-        break;
-      case UNAVAILABLE:
-        code = Status.Code.UNAVAILABLE;
-        break;
-      case UNIMPLEMENTED:
-        code = Status.Code.UNIMPLEMENTED;
-        break;
-      case UNKNOWN:
-        code = Status.Code.UNKNOWN;
-        break;
-      default:
-        code = Status.Code.UNKNOWN;
-    }
-    Throwable cause = (e.getCause() != null) ? e.getCause() : e;
```

Comment:
@ggezer why do we use the cause instead of the original message?

Reply:
The original message will be the one from coming from gRPC. We want a message that is less verbose because we show it directly to clients.

Reply:
OK, I talked for an earlier state I think. For this particular place, there is no clear answer. Earlier, with thrift, we were able to return the status exception along with its inner cause. Some tests were validating the inner exception and some were just the outer exception. This particular logic here is to maintain that behaviour with gRPC model where you're not allowed to return an inner exception.